socs/agents/hwp_supervisor/agent.py

```diff
@@ -138,6 +138,58 @@ def update(self):
         }
 
 
+@dataclass
+class ACUState:
+    """
+    Class containing ACU state information.
+
+    Args
+    ------
+    instance_id : str
+        Instance ID of ACU agent
+    min_el : float
+        Minimum elevation allowed before restricting spin-up [deg]
+    max_el : float
+        Maximum elevation allowed before restricting spin-up [deg]
+    max_time_since_update : float
+        Maximum time since last update before restricting spin-up[sec]
+
+    Attributes
+    ------------
+    el_current_position : float
+        Current el position [deg]
+    el_commanded_position : float
+        Commanded el position [deg]
+    el_current_velocity : float
+        Current el velocity [deg/s]
+    last_updated : float
+        Time of last update [sec]
+    """
+    instance_id: str
+    min_el: float
+    max_el: float
+    max_time_since_update: float
+
+    el_current_position: Optional[float] = None
+    el_commanded_position: Optional[float] = None
+    el_current_velocity: Optional[float] = None
+    last_updated: Optional[float] = None
+
+    def update(self):
+        op = get_op_data(self.instance_id, 'monitor')
+        if op['status'] != 'ok':
+            return
+
+        d = op['data'].get("StatusDetailed")
+        if d is None:
+            return
+
+        self.el_current_position = d['Elevation current position']
+        self.el_commanded_position = d['Elevation commanded position']
+        self.el_current_velocity = d['Elevation current velocity']
+        self.last_updated = time.time()
```

Comment:
The timestamp should come from the ACU dataset.  Or at least be rejected if doesn't make sense.

I acknowledge the ACU session data isn't making that very easy... can you change this to
```
last_updated = d.get('timestamp_agent')
```
and we'll get that field added into the agent session.data separately?

Reply:
Yep! Will `timestamp_agent` sometimes not exist / should I handle None values by just not changing `last_updated`?

Reply:
Right, sorry, I meant that you could then replace None with time.time().  That's a compromise until the ACU agent is updated.  Once ACU agent is updated, I would expect timestamp_agent to be either absent (during startup) or a float.